### PR TITLE
User guide initial draft

### PR DESCRIPTION
**Pull request by jakethekoenig**
* A getting started guide with installation + basic usage info
* A Context section with more details about how to manipulate and inspect LLM context.
* usage is renamed to Command Line Arguments
* Commands and Configuration sections are started as stubs
* internal is renamed to Developer Guide
* Copyright bump

### CODE REVIEW

**Comment by jakethekoenig** [comment]
You can build the docs with:
```
cd docs
sphinx-build -M html source build
```
In a future PR we'll have to make a github action that does that and uploads the output somewhere (or use readthedocs whose docs I need to read)

File: docs/source/user/context.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Context
+=======
+
+An LLM is only as good as its context. Therefore we provide a number of ways to manually manage and inspect context as well as a few tools to attempt to automatically select a context.
```
Comment (granawkins) [inline docs/source/user/context.rst:4]: "Context refers to portions of your code/files which are sent to the LLM along with your question/task, adn an LLM is only as good as its context."

Anchoring the word 'context' up-front might be helpful for some

File: docs/source/user/context.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Files can be manually added to context as a command line argument when starting mentat or with the :code:`/include` command during a session. Files can be removed from context with the :code:`/exclude` command. Mentat always puts all included files into the system message sent to the LLM so you probably don't want to start mentat with :code:`mentat .`. If you do want mentat to intelligently select the context from your prompt you should run :code:`mentat -a` and mentat will build its own context. For more see :ref:`auto context`.
+
+You can specify line ranges to add only a subset of a file to context. For example :code:`/include README.md:1-5,10-20` would add the first 5 and 10th to 20th lines to the LLMs context. You can see a summary of what is in context with the :code:`/context` command.
```
Comment (granawkins) [inline docs/source/user/context.rst:8]:
```suggestion
You can specify line ranges to add only a subset of a file to context by adding the starting line (inclusive) and ending line (exclusive) to the path. For example :code:`/include README.md:1-5,10-20` would add lines 1, 2, 3 and 4 and 10th to 19th lines to the LLMs context. You can see a summary of what is in context with the :code:`/context` command.
```
Just confirmed that the second number is in fact exclusive.

**Review by granawkins — APPROVED** [review]
Looks good - I ran `make html` locally and read through the output. I'd like to beef up the auto-context section a bit but that can be later, this is a good start.

**Comment by jakethekoenig** [comment]
> Looks good - I ran `make html` locally and read through the output. I'd like to beef up the auto-context section a bit but that can be later, this is a good start.

Thanks for the quick approval. Agree the context section could be improved. I think it's good to get a skeleton out there though to make it easier to improve in parallel. I'm hoping to have something we're happy hosting (probably on read the docs) by early next week.

**Pull request merged**